apm-agent-core/src/main/java/co/elastic/apm/agent/logging/LoggingConfiguration.java

```diff
@@ -159,8 +159,38 @@ public void assertValid(Boolean value) {
         })
         .buildWithDefault(false);
 
+    private final ConfigurationOption<Boolean> logShadingEnabled = ConfigurationOption.booleanOption()
```

Comment:
Let's make it clear in the docs that log shading is experimental and may be removed in future releases and that we recommend replacing the plain-text formatters with ECS formatters (elastic/apm#373).
Should we make the log shading options internal?

Reply:
The new `log_ecs_reformatting` config will be marked experimental, so it will give us time to gather feedback on how useful these options are without worrying about removing them in the future.
My next task in this area will be the addition of the `OVERRIDE` option, where I can add a note that it is preferred if we decide so.